ext/jruby/org/jruby/ext/strscan/RubyStringScanner.java

```diff
@@ -556,6 +557,47 @@ public IRubyObject peep(ThreadContext context, IRubyObject length) {
         return peek(context, length);
     }
 
+    @JRubyMethod(name = "scan_integer")
+    public IRubyObject scan_integer(ThreadContext context) {
+        final Ruby runtime = context.runtime;
+        check(context);
+        clearMatched();
+
+        if (!str.getEncoding().isAsciiCompatible()) {
+            throw getRuntime().newEncodingCompatibilityError("ASCII incompatible encoding: " + str.getEncoding());
+        }
+
+
+        ByteList bytes = str.getByteList();
+        int curr = this.curr;
+
+        int bite = bytes.get(curr);
```

Comment:
`byte`?

Reply:
Thanks for asking that, I now re-learned that in Java, there are certain reserved words: https://en.wikipedia.org/wiki/List_of_Java_keywords and `byte` is one of them.

Reply:
!!!